protocol/x/listing/keeper/listing.go

```diff
@@ -90,3 +91,39 @@ func (k Keeper) CreateClobPair(
 
 	return clobPair.Id, nil
 }
+
+// Function to wrap the creation of a new perpetual
+// Note: This will only list long-tail/isolated markets
+// TODO: Complete implementation and add tests pending marketmap decoding functions and testutils
+func (k Keeper) CreatePerpetual(
+	ctx sdk.Context,
+	marketId uint32,
+	ticker string,
+) (perpetualId uint32, err error) {
+	perpetualId = k.PerpetualsKeeper.AcquireNextPerpetualID(ctx)
+
+	// Calculate atomic resolution from market map reference price
+	var atomicResolution int32
+	//marketMapDetails, err := k.MarketMapKeeper.GetMarket(ctx, ticker)
+	//if err != nil {
+	//	return 0, err
+	//}
+	// TODO: get reference price once market map decoding functions are available
+
+	// Create a new perpetual
```

Comment:
nit: Make this also a TODO? Isn't `atomicResolution` 0 here?

Reply:
Like have this code, but commented?

Reply:
I can't have the code because the utility doesn't exist yet